tsnet/tsnet.go

```diff
@@ -536,7 +540,7 @@ func (s *Server) Listen(network, addr string) (net.Listener, error) {
 		return nil, err
 	}
 
-	key := listenKey{network, host, uint16(port)}
+	key := listenKey{network, "", uint16(port)}
```

Comment:
how about we keep it the host but try more matches on the lookup side?

Reply:
That makes sense—but I'm not sure what keys we would use. I guess we could use `s.hostname`, since at this point it's been filled in by `start()`. Are there other possible values we need to consider?

Reply:
I ask because it seems like an incoming connection could have lots of different spellings: The bare hostname, the fully-qualified hostname, some IP address, perhaps aliases?

Reply:
From most specific to least specific:

* "IP:port" (or "[ip]:port" for IPv6)
* "hostname:port"
* ":port"

Then could return an error if it's not one of those?

But then you'd expect an error if they listen on "tcp6" but give an IPv4 or vice versa.

But I don't think anything but "tcp" even works today.

Reply:
To get that, I think we have to call `conn.LocalAddr()` and re-parse the string, and then hold that up against what all those addresses are supposed to be. Presumably those are assigned by the coordination server—but they don't seem to be recorded in the Server type so I guess we'd have to either look them up, or update the setup to save them. 

It seems like a simpler solution is to simply disallow (and report an error for) non-empty hostnames in Listen: They already don't work now, so we wouldn't be breaking any working software—and getting an error would be an improvement over the silent failure we have with the current situation.